org-roam.el

```diff
@@ -503,6 +503,32 @@ This uses the templates defined at `org-roam-capture-templates'."
   (let ((time (org-read-date nil 'to-time nil "Date:  ")))
     (org-roam--file-for-time time)))
 
+(defun org-roam--move-days (days)
+  "If the current note has a title in date format yyyy-mm-dd, add
+DAYS days (can be negative) and open the note for the resulting
+new date.  If that note does not exist, it will be created."
+  (let* ((title (car (plist-get (org-export-get-environment) :title))))
+    (unless (string-match-p "^[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]$" title)
```

Comment:
Here it would be good to use `org-roam-date-title-format` to parse the string. But it appears there is no standard elisp function for parsing string following a specific format?  I must be overlooking something.

Reply:
[Someone wrote](https://emacs.stackexchange.com/questions/57460/check-whether-a-string-is-a-date-in-a-specific-format-e-g-y-m-d) an Elisp wrapper for strptime(3) that could be used here.